pandas/_libs/groupby_helper.pxi.in

```diff
@@ -584,7 +584,10 @@ def group_rank_{{name}}(ndarray[float64_t, ndim=2] out,
 
         if pct:
             for i in range(N):
-                out[i, 0] = out[i, 0] / grp_sizes[i, 0]
+                if out[i, 0] != out[i, 0] or out[i, 0] == NAN:
```

Comment:
Can this all be simplified to just `if out[i, 0] != NAN:`?

Reply:
I don't think those are semantically equivalent.  Did you mean `out[i, 0] == NAN` ?

Reply:
What I meant was only do the division if `out[i, 0] != NAN` otherwise leave as is

Reply:
Ah, gotcha! 🙂 Let's try it and see what happens.

Reply:
@WillAyd : Good idea, but unfortunately, the tests I added disagree with it.  You need both conditionals when checking.  Thus, this code needs to stay as is.

Reply:
Hmm OK I got you. Is it a particular type that's failing? 

For some reason I thought some of the work @realead was doing was supposed to remove the need for comparisons like `out[i, 0 ] != out[i, 0]` to figure out if a value was NA though it's entirely possible I have misunderstood that

Reply:
The "right" handling of NA would only apply to algorithms using hash-map, which is here not the case here if I see it correctly.

Reply:
can you add a comment on what is going on here

Reply:
Yep, done.